### ISSUE TITLE
nodeenv --list is raising TypeError

### ISSUE THREAD

**Issue by syndbg** [ekalinin/nodeenv#117]
When installed globally with `sudo pip install nodeenv`, `nodeenv --list` raises an exception.

This is due to `pkg_resources.parse_version` method having different return types in older versions. 

![nodeenv_bug](https://cloud.githubusercontent.com/assets/6031925/6599860/d253a2b0-c814-11e4-94d6-9a9d79cdf27c.png)

Possible hotfix solutions are:
* write a helper function to parse strings,
* write our own Version type :-1: 

But atm I'm working on finding out which version of `setuptools` introduced the newer (currently used in master) Version type that parses version strings and rely on it if `setuptools` is >= that version.

I'll combine that with one of two approaches:
* the raw string representation we get with regex pattern. However ascending sorting will be `0.1.0, 0.10.0, 0.9.0`,
*  the before mentioned helper function

I think broken ascending sort is acceptable as a hotfix.

PR will be submitted soon-ish. :8ball:

**Comment by ekalinin** [comment]
Hi @syndbg 
Thanks for report.
I'm getting the same error on Ubuntu (just checked):
```
➥ cat /etc/issue
Ubuntu 14.10 \n \l
➥ uname -i
x86_64
➥ LANG=C sudo aptitude show python-pip | grep Ver
Version: 1.5.6-2
➥ sudo aptitude install python-pip -y
➥ sudo pip install nodeenv
➥ nodeenv --version
0.13.0
➥ nodeenv --list
Traceback (most recent call last):
  File "/usr/local/bin/nodeenv", line 9, in <module>
    load_entry_point('nodeenv==0.13.0', 'console_scripts', 'nodeenv')()
  File "/usr/local/lib/python2.7/dist-packages/nodeenv.py", line 820, in main
    print_node_versions()
  File "/usr/local/lib/python2.7/dist-packages/nodeenv.py", line 734, in print_node_versions
    versions = get_node_versions()
  File "/usr/local/lib/python2.7/dist-packages/nodeenv.py", line 727, in get_node_versions
    return [v.public for v in sorted_versions]
AttributeError: 'tuple' object has no attribute 'public'
```

**Comment by ekalinin** [comment]
More interesting that all tests inside virtualenv works well (``make tests`` in the root if the project).

**Comment by syndbg** [comment]
@ekalinin tl;dr That's due to apt-get packages being outdated. No surprise since it's an user-unfriendly ecosystem for package distribution. 

## However, in details:

### `pkg-resources` version that supports the newer `parse_version` is 8.0 and newer.

See the tag 8.0 and `pkg_resources.py`
https://bitbucket.org/pypa/setuptools/commits/7ea0e7498e4d


### apt-get's `python-pkg-resources` and `python3-pkg-resources` version in Ubuntu: `5.5.1`

```
▶ apt-cache showpkg python-pkg-resources
Package: python-pkg-resources
Versions:
5.5.1-1 (/var/lib/apt/lists/archive.ubuntu.com_ubuntu_dists_utopic_main_binary-amd64_Packages) (/var/lib/dpkg/status)
...
Provides:
5.5.1-1 - python2.7-setuptools 

```

```
▶ apt-cache showpkg python3-pkg-resources
Package: python3-pkg-resources
Version: 5.5.1-1
...
Provides:
python3.3-setuptools

```


### `python3-virtualenv`, `python-virtualenv`, `python3-pip` and `python-pip` depend on out-dated `pkg-resources`

```
▶ apt-cache show python3-virtualenv
Package: python3-virtualenv
...
Source: python-virtualenv
Version: 1.11.6-2
Depends: python-pip-whl, python-setuptools-whl, python3, **python3-pkg-resources**, python3:any (>= 3.3.2-2~)
``` 

```
▶ apt-cache show python-virtualenv
Package: python-virtualenv
...
Version: 1.11.6-2
Depends: python, python-pip-whl, **python-pkg-resources**, python-setuptools-whl
```

```
▶ apt-cache show python-pip       
Package: python-pip
...
Version: 1.5.6-2
Depends: ca-certificates, python-colorama, python-distlib, python-html5lib, **python-pkg-resources**, python-requests, python-setuptools (>= 0.6c1), python-six, python:any (>= 2.7.5-5~), python, python:any (<< 2.8)
```

```
▶ apt-cache show python3-pip      
Package: python3-pip
...
Version: 1.5.6-2
Depends: ca-certificates, python3-colorama, python3-distlib, python3-html5lib, **python3-pkg-resources**, python3-requests, python3-setuptools, python3-six, python3, python3:any (>= 3.3.2-2~)
```


**But you ask why does it work sometimes in a sourced virtualenv?**
As seen in apt pkg infos, older pip versions `1.X` are using older `pkg_resources`. However I'm too lazy to check which exact pip version comes with `pkg_resources>=8.0`.

But if you have updated your pip to come with a newer `pkg_resources` by doing `pip install pip -U`, you're good to go. :+1: 

Tests pass with pip versions as `6.X`, but that's a minority.



Still the PR I submitted handles it.

**Comment by ekalinin** [comment]
Fixed by #118
